File: code/py/src/leetcode/done/until20231221/p867.py
from rockyutil.leetcode import *


class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        m, n = len(matrix), len(matrix[0])
        return [[matrix[_j][_i] for _j in range(m)] for _i in range(n)]

File: code/kt/src/onlinejudge/leetcode/solved/P867.kt
package onlinejudge.leetcode.solved

class P867 {
    fun transpose(matrix: Array<IntArray>): Array<IntArray> {
        val ans: Array<IntArray> = Array(matrix[0].size) { IntArray(matrix.size) }
        for (i in matrix.indices) {
            for (j in matrix[i].indices) {
                ans[j][i] = matrix[i][j]
            }
        }
        return ans
    }
}
